pkg/validation/apiextensions/v1/composition/admission_handler.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package composition
```

Comment:
AFAIR, the motivation behind having these packages under `pkg` instead of `internal` is to have the validation logic importable from other tools, right? Considering most of the validation logic added as methods to the types in `apis` directory, I am wondering if we still need a separate importable package if we move helpers like `WrapFieldError()` to crossplane-runtime (or crossplane-tools or a seperate repo).